### DISCUSSION TITLE
commandOnWithTimedOff not available via zigbee2mqtt?

### DISCUSSION THREAD

**Discussion by busywait** [Koenkk/zigbee2mqtt#19691] [General]
I would like to use the commandOnWithTimedOff command to have my bulb turn on for only a set amount of time before it turns itself off.  I think that Z2M only allows me to pass state "on" or state "off", is that correct?

I have an IKEA bulb ([/devices/LED2035G10.html](https://www.zigbee2mqtt.io/devices/LED2035G10.html#ikea-led2035g10)) that is bound to an IKEA PIR ([/devices/E1525_E1745.html](https://www.zigbee2mqtt.io/devices/E1525_E1745.html#ikea-e1525%252Fe1745))

The PIR sends messages like 
`Zigbee message from 'device', type 'commandOnWithTimedOff', cluster 'genOnOff', data '{"ctrlbits":0,"offwaittime":0,"ontime":600}' from endpoint 1 with groupID 123`

The bulb responds by turning on for the specified time (measured in 1/10s of a second), and then turning off.

I would like to be able to send a similar message to the bulb over MQTT from Home Assistant so that my aqara door contact sensor can also turn on the bulb with the same "timed off" behaviour.

Using Home Assistant, Mosquito and Zigbee2mqtt I have seen that I can turn the bulb on or off by publishing messages
`Topic
zigbee2mqtt/Front Porch Bulb/set

Payload
{"state": "on"}`

I can't see a way to use onWithTimedOff or the ontime attributes as shown in the log example above.

 Is this a reasonable feature request, or have a missed a trick?

**Comment by okastl** [comment]
> Payload
> {"state": "on"}`

Try his:
`{"state":"on", "on_time":3.0}
`
The lamp should turn off after 3 seconds.

Some devices need "off_wait_time" to be set, too:
`{"state":"on", "on_time":3.0, "off_wait_time":3.0 }
`

**Reply by busywait** [reply]
Yes! Thank you :)

The first variation works for me (and so does the second, but I don't need the guard time with my PIR).

**Reply by busywait** [reply]
I'm curious how did you know? I didn't find these documented anywhere - I might have been bad at looking. Now that you've shown the correct syntax it was easy to find this github issue where the attributes and usage are described [github.com/Koenkk/zigbee-herdsman-converters/pull/2326](https://github.com/Koenkk/zigbee-herdsman-converters/pull/2326)

**Reply by okastl** [reply]
I honestly don't remember how I know. I did dig through z2m source code a while ago to write experimental device extensions, maybe I just found it.
